addons/pos_loyalty/models/pos_order.py

```diff
@@ -144,6 +147,20 @@ def confirm_coupon_programs(self, coupon_data):
             'coupon_report': coupon_per_report,
         }
 
+    def _check_existing_loyalty_cards(self, coupon_data):
+        coupon_key_to_modify = []
+        for coupon_id, coupon_vals in coupon_data.items():
+            partner_coupons = self.env['loyalty.card'].search(
+                [('partner_id', '=', coupon_vals.get('partner_id', False)), ('program_type', '=', 'loyalty')])
```

Comment:
We should not consider the loyalty.card's that doesn't have partner_id, so maybe only make this check if 'partner_id' is present.